### PR TITLE
Refactor boilerplate to use SqlConnectionStringBuilder instead of relying on objects for testing connection strings

### PR DESCRIPTION
**Pull request by Stefanie899**
- [NA] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+
+namespace AndcultureCode.GB.Infrastructure.Data.SqlServer.Migrations
+{
+    public partial class GBFlattenedMigration : FlattenedMigration
+    {
+        public const string InitialMigrationId = "20200603183650_InitialCreate";
```
Comment (wintondeshong) [inline dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs:14]: Nitpick: Region

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        public override List<string> FlattenedMigrations
+        {
+            get
+            {
+                return _flattenedMigrations;
+            }
+        }
+
+        public override DbContext Context
```
Comment (wintondeshong) [inline dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs:42]: Suggestion: Consider use of expression bodies here

```csharp
public override DbContext Context
{
    get => _context = _context ?? new GBApiContext();
}
```

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        #endregion Private
+
+        #region Overrides
+
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+        }
+
+        public override List<string> FlattenedMigrations
```
Comment (wintondeshong) [inline dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs:34]: Suggestion: Consider expression bodies here

```csharp
public override List<string> FlattenedMigrations
{
    get => _flattenedMigrations;
}
```

File: dotnet/api/Presentation/Cli/Command.cs [hunk 4]
```diff
@@ -75,8 +75,8 @@ protected GBApiContext _testDatabaseContext
             {
                 if (_cachedTestDatabaseContext == null)
                 {
-                    var connection = _configurationRoot.GetTestDatabaseConnection();
-                    _cachedTestDatabaseContext = new GBApiContext(connection, null);
+                    var connectionStringBuilder = _configurationRoot.GetTestDatabaseConnectionStringBuilder();
+                    _cachedTestDatabaseContext = new GBApiContext(connectionStringBuilder.ConnectionString, null);
```
Comment (wintondeshong) [inline dotnet/api/Presentation/Cli/Command.cs:79]: Suggestion: Consider alternative constructor for GBApiContext or using named parameters to describe the `null`

File: dotnet/api/Presentation/Cli/appsettings.cli.json [hunk 5]
```diff
@@ -1,11 +1,5 @@
...
-        "Api": {
-            "AdditionalParameters": "MultipleActiveResultSets=true",
-            "Database": "GravityBootsApi-Test",
-            "Datasource": ".\\",
-            "Password": "passw0rd!",
-            "UserId": "gbuser"
-      }
+        "Api": "Data Source=.; Database=GravityBootsApi-Test; user id=gbuser; password=passw0rd!; MultipleActiveResultSets=true;"
     }
 }
```
Comment (wintondeshong) [inline dotnet/api/Presentation/Cli/appsettings.cli.json:5]: Nitpick: Newline to end file

File: dotnet/api/Presentation/Cli/appsettings.cli.json [hunk 6]
```diff
@@ -1,11 +1,5 @@
 {
     "ConnectionStrings": {
-        "Api": {
-            "AdditionalParameters": "MultipleActiveResultSets=true",
-            "Database": "GravityBootsApi-Test",
-            "Datasource": ".\\",
-            "Password": "passw0rd!",
-            "UserId": "gbuser"
-      }
+        "Api": "Data Source=.; Database=GravityBootsApi-Test; user id=gbuser; password=passw0rd!; MultipleActiveResultSets=true;"
```
Comment (wintondeshong) [inline dotnet/api/Presentation/Cli/appsettings.cli.json:3]: 👍

File: dotnet/api/Tests/Business/Conductors.Tests/appsettings.json [hunk 7]
```diff
@@ -1,11 +1,5 @@
...
-        "Api": {
-            "AdditionalParameters": "MultipleActiveResultSets=true",
-            "Database": "GravityBootsApi-Test",
-            "Datasource": ".\\",
-            "Password": "passw0rd!",
-            "UserId": "gbuser"
-        }
+        "Api": "Data Source=.; Database=GravityBootsApi-Test; user id=gbuser; password=passw0rd!; MultipleActiveResultSets=true;"
     }
 }
```
Comment (wintondeshong) [inline dotnet/api/Tests/Business/Conductors.Tests/appsettings.json:5]: Nitpick: newline to end file

File: dotnet/api/Tests/Presentation/Worker.Tests/appsettings.json [hunk 8]
```diff
@@ -1,11 +1,5 @@
...
-        "Api": {
-            "AdditionalParameters": "MultipleActiveResultSets=true",
-            "Database": "GravityBootsApi-Test",
-            "Datasource": ".\\",
-            "Password": "passw0rd!",
-            "UserId": "gbuser"
-        }
+        "Api": "Data Source=.; Database=GravityBootsApi-Test; user id=gbuser; password=passw0rd!; MultipleActiveResultSets=true;"
     }
 }
```
Comment (wintondeshong) [inline dotnet/api/Tests/Presentation/Worker.Tests/appsettings.json:5]: Nitpick: newline to end file

File: dotnet/api/Tests/Testing/Fixtures/DatabaseFixture.cs [hunk 9]
```diff
@@ -18,8 +18,8 @@ public abstract class DatabaseFixture : IDisposable
         #region Properties
 
         private IConfigurationRoot _configuration { get; set; }
-        public GBApiConnection Connection { get; set; }
-        public IContext Context => new GBApiContext(Connection, null);
+        public SqlConnectionStringBuilder ConnectionStringBuilder { get; set; }
+        public IContext Context => new GBApiContext(ConnectionStringBuilder.ConnectionString, null);
```
Comment (wintondeshong) [inline dotnet/api/Tests/Testing/Fixtures/DatabaseFixture.cs:22]: Suggestion: Same as above about named params or alternative constructor

File: dotnet/api/Tests/Testing/Fixtures/DatabaseFixture.cs [hunk 10]
```diff
@@ -116,7 +116,7 @@ public void CleanDatabaseTables(SqlConnection sqlConnection, List<string> tableN
         /// <returns></returns>
         public Context GetNewContext()
```
Comment (wintondeshong) [inline dotnet/api/Tests/Testing/Fixtures/DatabaseFixture.cs:117]: Suggestion: Consider use of expression bodies

**Review by wintondeshong — APPROVED** [review]
Nice work @Stefanie899 - Thank you for giving this area some much needed love 👍 -- all my feedback is food for thought so you have my approval

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs (hunk 2 above)
Comment (Stefanie899) [reply dotnet/api/Infrastructure/Data.SqlServer/Migrations/GBFlattenedMigration.cs:42]: Good call, I need to use these more often.

**Pull request merged**